src/main/java/org/broadinstitute/hellbender/tools/exome/AllelicCountCollection.java

```diff
@@ -64,12 +65,34 @@ public void add(final SimpleInterval interval, final int refReadCount, final int
         counts.add(new AllelicCount(interval, refReadCount, altReadCount));
     }
 
+    /**
+     * Adds (interval, reference count, alternate count, reference nucleotide, alternate nucleotide) to respective lists.
+     * @param interval      site in 1-based format
+     * @param refReadCount  number of reads at site matching the reference
+     * @param altReadCount  number of reads at site different from the reference
+     * @param refNucleotide the ref nucleotide
+     * @param altNucleotide the alt nucleotide
+     */
+    public void add(final SimpleInterval interval, final int refReadCount, final int altReadCount,
+                    final Nucleotide refNucleotide, final Nucleotide altNucleotide) {
+        counts.add(new AllelicCount(interval, refReadCount, altReadCount, refNucleotide, altNucleotide));
+    }
+
     /** Returns an unmodifiable view of the list of AllelicCounts.   */
     public List<AllelicCount> getCounts() {
         return Collections.unmodifiableList(counts);
     }
 
     /**
+     * @return a map from SimpleIntervals in counts to their integer index
+     */
+    public Map<SimpleInterval, Integer> getSimpleIntervalToIndexMap() {
+        Map<SimpleInterval, Integer> intervalToIndexMap = new HashMap<>();
```

Comment:
First I would make ```intervalToIndexMap``` ```final``` and give it a size estimate in the constructor as you know how many elements it is going to have.